Helper/Invoice.php

```diff
@@ -249,14 +261,19 @@ public function handleCaptureWebhook(Order $order, Notification $notification):
     {
         $invoiceFactory = $this->adyenInvoiceFactory->create();
         $adyenInvoice = $this->adyenInvoiceResourceModel->getAdyenInvoiceByCaptureWebhook($order, $notification);
+        $fullAmountCaptured = $this->adyenDataHelper->originalAmount($notification->getAmountValue(), $notification->getAmountCurrency()) >= $order->getBaseGrandTotal();
 
-        if (is_null($adyenInvoice)) {
-            throw new \Exception(sprintf(
-                'Unable to find adyen_invoice linked to original reference %s, psp reference %s, and order %s',
-                $notification->getOriginalReference(),
-                $notification->getPspreference(),
-                $order->getIncrementId()
-            ));
+        if (is_null($adyenInvoice) && $order->canInvoice()) {
+                if($fullAmountCaptured) {
+                    $this->createInvoiceFromWebhook($order, $notification);
+                } else {
+                    $order->addStatusHistoryComment(__(sprintf(
+                        'Partial %s webhook notification w/amount %s %s was processed, no invoice created. Please create offline invoice.',
```

Comment:
Since we don't have an `$adyenInvoice` in this `else` branch, should it `return` after adding this comment?

Reply:
Throwing an exception here as well, as the return type of the `handleCaptureWebhook()` should be `AdyenInvoice`